api/certs/cert.go

```diff
@@ -202,11 +226,35 @@ func prepareCerts(wssCertExists, httpsCertExists bool, wssCertReady, httpsCertRe
 	}
 }
 
-func ChangeToDefault(https, wss bool) {
-	defaultDomain, err := util.GetDefaultDomainFromIP(config.Parameters.CertDomainName, config.Parameters.DefaultTlsDomainTmpl)
-	if err != nil {
-		log.Error(err)
+func SetDefaultDomain(https, wss bool) (bool, error) {
+	var defaultDomain string
+	var err error
+	var defaultCertAvailable bool
+	for _, d := range config.Parameters.DefaultTlsDomainTmpl {
+		defaultDomain, err = util.GetDefaultDomainFromIP(config.Parameters.CertDomainName, d)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		ok, err := domainDNSCheck(defaultDomain)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		if ok {
+			_, _, certFile, keyFile := GetACMEFileNames(defaultDomain, config.Parameters.CertDirectory)
+			_, err = VerifyLocalCertificate(certFile, keyFile, defaultDomain)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+		}
+		break
+	}
+	if len(defaultDomain) == 0 {
```

Comment:
`defaultDomain` might not be empty if `domainDNSCheck` or `VerifyLocalCertificate` fail